test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Options/PackageSourcesPageTests.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Utilities.UnifiedSettings;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio.Options;
+using NuGet.VisualStudio;
+using Test.Utility;
+using Xunit;
+
+namespace NuGet.PackageManagement.VisualStudio.Test.Options
+{
+    [Collection(MockedVS.Collection)]
+    public class PackageSourcesPageTests : NuGetExternalSettingsProviderTests<PackageSourcesPage>
+    {
+        private IEnumerable<PackageSource>? _packageSources;
+
+        public PackageSourcesPageTests(GlobalServiceProvider sp)
+        {
+            sp.Reset();
+            NuGetUIThreadHelper.SetCustomJoinableTaskFactory(ThreadHelper.JoinableTaskFactory);
+            _packageSources = Enumerable.Empty<PackageSource>();
+        }
+
+        protected override PackageSourcesPage CreateInstance(VSSettings? vsSettings)
+        {
+            TestPackageSourceProvider packageSourceProvider = new(_packageSources);
+
+            return new PackageSourcesPage(vsSettings!, packageSourceProvider);
+        }
+
+        [Fact]
+        public async Task GetValueAsync_WhenNuGetConfigHasInvalidSource_ReturnsFailureResultTaskAsync()
```

Comment:
`PackageSourceValidatorTests` cover most scenarios. Here, I wanted to emphasize how USX will handle package sources errors from an externally edited `nuget.config`.